pkg/policy/l4.go

```diff
@@ -325,26 +328,34 @@ func (l4 *L4Filter) ToMapState(policyOwner PolicyOwner, direction trafficdirecti
 		//    redirect    allow     no redirect deny    yes  (deny takes precedence)
 		//    no redirect allow     no redirect allow   yes  (same effect)
 		//    no redirect allow     no redirect deny    yes  (deny takes precedence)
-		//    no redirect deny      no redirect allow   no   (deny takes precedence)
+		// ----------------------------------------------------------------------------------
+		// #) no redirect deny      no redirect allow   no   (deny takes precedence)
+		// ----------------------------------------------------------------------------------
 		//    no redirect deny      no redirect deny    yes  (same effect)
 		//    redirect    allow     redirect    allow   yes  (same effect)
-		//    no redirect allow     redirect    allow   yes  (must redirect if L4-only redirects)
-		//    no redirect deny      redirect    allow   no   (deny takes precedence)
+		// ** no redirect allow     redirect    allow   yes  (must redirect if L4-only redirects)
+		// ----------------------------------------------------------------------------------
+		// #) no redirect deny      redirect    allow   no   (deny takes precedence)
+		// ----------------------------------------------------------------------------------
 		//
 		// have wildcard?        this is a L3L4 key?
-		isDeny := l7 != nil && l7.IsDeny
+		isDenyRule := rule != nil && rule.IsDeny
 		wildcardL7PolicyIsDeny := wildcardL7Policy != nil && wildcardL7Policy.IsDeny
 
-		if l4.wildcard != nil && cs != l4.wildcard &&
-			// not the "no" case marked with *)?
-			!(l7.IsRedirect() && !wildcardL7Policy.IsRedirect() && !wildcardL7PolicyIsDeny &&
-				// l3/l4 allow OR l4-only deny
-				(!isDeny || wildcardL7PolicyIsDeny)) {
+		ruleHasWildcardCacheSelector := l4Filter.wildcard != nil
+		isWildcardL7Rule := cs == l4Filter.wildcard
+
+		// eliminating the first 3 lines from the table above
+		if ruleHasWildcardCacheSelector && !isWildcardL7Rule &&
+			// eliminating the "Skip 'no'" case for the line marked with *)
+			!(rule.IsRedirect() && !wildcardL7Policy.IsRedirect() && !wildcardL7PolicyIsDeny) &&
+			// eliminating the "Skip no'" cases for the lines marked with #)
```

Comment:
Thanks for the comments. I'm not convinced I understand this logic well enough to make any commentary, but I at least agree it matches the table.